storehaus-mongodb/src/main/scala/com/twitter/storehaus/mongodb/MongoStore.scala

```diff
@@ -86,31 +89,30 @@ class MongoStore[K: MongoValue, V: MongoValue: Manifest] (
 
   override def get(key: K): Future[Option[V]] = futurePool {
     col.findOne(MongoDBObject(keyName -> key)).flatMap { valueObject => {
-        getValue(valueObject)
+        getValue(valueObject).toOption
       }
     }
   }
 
-  protected def getValue(valueObject: MongoDBObject): Option[V] = {
+  protected def getValue(valueObject: MongoDBObject): Try[V] =
     valueObject.getAs[V](valueName) match {
-      case None => {
-        if (valueObject.get(valueName) == None) None
-        else throw new ClassCastException("Cannot convert %s to %s".format(valueObject.get(valueName).getClass(), manifest[V]))
-      }
-      case Some(value) => {
-        if (getManifest().isInstance(value)) Some(value)
-        else throw new ClassCastException("Cannot convert %s to %s".format(value.getClass(), manifest[V]))
-      }
+      case None =>
+        if (valueObject.get(valueName).isEmpty) Throw(new Exception("Value not found"))
```

Comment:
I think this should be `Return(None)` on this branch, which is to say it was really empty.